profile/src/main/java/rdx/works/profile/data/model/Profile.kt

```diff
@@ -58,41 +59,45 @@ data class Profile(
         )
     )
 
-    /**
-     * Temporarily the only factor source that the user can use to create accounts/personas.
-     * When new UI is added that allows the user to import other factor sources
-     * (like an Olympia device factor source), we will need to revisit this.
-     *
-     * NOTE that this factor source will always be used when creating the first account.
-     */
-    val babylonMainDeviceFactorSource: DeviceFactorSource
-        get() = factorSources
-            .filterIsInstance<DeviceFactorSource>()
-            .firstOrNull { deviceFactorSource ->
-                deviceFactorSource.isBabylon && deviceFactorSource.common.flags.any { it == FactorSourceFlag.Main }
-            } ?: factorSources
-            .filterIsInstance<DeviceFactorSource>()
-            .first { deviceFactorSource ->
-                deviceFactorSource.isBabylon
-            }
-
-    val babylonDeviceFactorSourceExist: Boolean
-        get() = factorSources
-            .filterIsInstance<DeviceFactorSource>()
-            .any {
-                it.isBabylon
-            }
-
     companion object {
         fun init(
             id: String,
             deviceInfo: DeviceInfo,
             creationDate: Instant,
-            gateways: Gateways = Gateways.preset
+            gateways: Gateways = Gateways.preset,
+        ): Profile {
+            val appPreferences = AppPreferences(
+                transaction = Transaction.default,
+                display = Display.default,
+                security = Security.default,
+                gateways = gateways,
+                p2pLinks = listOf()
+            )
+
+            return Profile(
+                header = Header.init(
+                    id = id,
+                    deviceInfo = deviceInfo,
+                    creationDate = creationDate,
+                    numberOfNetworks = 0
+                ),
+                appPreferences = appPreferences,
+                factorSources = emptyIdentifiedArrayList(),
+                networks = emptyList()
+            )
+        }
+
+        fun initProfileWithRecoveredBDFS(
```

Comment:
Maybe `init` is also suited, or `initWithFactorSource`. I know that this is implying that it is a bdfs, but since the method does not check if it is one the name above would be better.